### ISSUE TITLE
received a hostname length error as part of magic castle deployment

### ISSUE THREAD

**Issue by edwins** [dun/munge#140]
Hello,

I was using [magic castle](https://github.com/ComputeCanada/magic_castle), and it turns out I received the following error (masking the actual hostname while keeping the length the same):
```
mgmt1.int.alma9-snap-vol-lg-mgmt-5.xxx180106.something-cloud.org munged[18658]: munged: Error: Failed to determine hostname: File name too long
```
Here is the munged version info for the host in question
```
# munged --version
munge-0.5.13 (2017-09-26)
```
This was running in Alma9. I thought I would be useful to report.

Thank you!

**Comment by dun** [comment]
Hi @edwins,

0.5.13 calls `gethostname()` with a buffer size of `MAXHOSTNAMELEN`. That's defined in `<sys/param.h>` to be 64. And since the hostname you listed above is exactly 64 bytes (plus 1 for the terminating NUL character), it exceeds that buffer and generates the error.

However, 0.5.14 changed that code with the introduction of the `--origin` option. `gethostname()` is now called with a buffer size of (`_POSIX_HOST_NAME_MAX` + 1). And `_POSIX_HOST_NAME_MAX` is defined in `<limits.h>` to be 255. Also, 0.5.14 doesn't treat a failure to resolve the hostname as a fatal error anymore.

Can you try running 0.5.14 or later and let me know if this problem persists? (It shouldn't.) 0.5.16 is the latest release. I tested the RPMs built from the release tarball on Alma9.

**Comment by dun** [comment]
Closing since this issue appears to be fixed in 0.5.14. Please follow up if this problem persists in the latest release.
